libs/wingsdk/src/shared/misc.ts

```diff
@@ -72,3 +73,15 @@ export async function runDockerImage({
 
   return { hostPort };
 }
+
+var _id = 0;
+export function inflightId(inflight?: IInflight) {
```

Comment:
I am wondering if perhaps we can just hash the contents of the inflight to generate a more stable id.

Using a global incrementing number means that multiple apps that run in the same process (eg for testing) will result in identifiers that are not deterministic (depends on the order of execution of the tests).

Alternatively, can we get rid of the id altogether somehow?

Reply:
This id is stable, or at least it's as stable (in wing) as our current id mechanism for inflight.

> if perhaps we can just hash the contents of the inflight to generate a more stable id.

The problem with using a content-based id is that we (currently) need to be able to uniquely identify inflights even if they have the same content.

> Using a global incrementing number means that multiple apps that run in the same process (eg for testing) will result in identifiers that are not deterministic

Yes and no. Due to all of this being synchronous and the preflight sandboxing we do, this might still work as expected. Additionally, for testing I think it's reasonable to expect isolation for globals like this (like how it is deterministically working in the wingsdk unit tests now).
I understand the concern though and I would prefer figure out an alternative solution.

> Alternatively, can we get rid of the id altogether somehow?

Maybe, I'm not sure. Part of my remaining TODOs is figuring out what to do with this thing, and getting rid of it would be my favorite option.